packages/debug/index.js

```diff
@@ -2,17 +2,9 @@
 
 module.exports = function debug(entity) {
   if (process.env.XMPP_DEBUG) {
-    entity.on('input', data => console.log('⮈ IN ', data))
-    entity.on('output', data => console.log('⮊ OUT', data))
-    ;['connect', 'open', 'authenticated', 'online', 'error', 'authenticate'].forEach(event => {
-      const prefix = (event === 'error' ? '❌   ' : '🛈   ')
-      entity.on(event, arg => {
-        if (arg === undefined || arg === null || typeof arg === 'function') {
-          console.log(prefix, event)
-        } else {
-          console.log(prefix, event, arg.toString())
-        }
-      })
-    })
+    entity.on('input', data => console.log('⮈', data))
+    entity.on('output', data => console.log('⮊', data))
+    entity.on('error', err => console.error('❌', err))
+    entity.on('status', (status, value) => console.log('🛈', status, value ? value.toString() : ''))
```

Comment:
I really like how status event simplifies this code. This really shows that folding separate events to one makes sense semantically.